### PR TITLE
update deprecated interface and fix bug not return when list pod failed in cronjob_controller.go

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:
remove some unused redundant code, and fix bug: when list pod failed, 
job still deleted but pod may still exist  in func `deleteJob`

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/cronjob/cronjob_controller.go [hunk 1]
```diff
@@ -263,9 +264,8 @@ func syncOne(sj *batchv2alpha1.CronJob, js []batchv1.Job, now time.Time, jc jobC
 		glog.V(4).Infof("No unmet start times for %s", nameForLog)
 		return
 	}
-	if len(times) > 1 {
-		glog.V(4).Infof("Multiple unmet start times for %s so only starting last one", nameForLog)
-	}
```
Comment (soltysh) [inline pkg/controller/cronjob/cronjob_controller.go]: Please change this back, we want to log only when more than one time was left behind.

**Review by soltysh — CHANGES_REQUESTED** [review]
One nit, do you think you could write a test case covering this part?

**Comment by hzxuzhonghu** [comment]
@soltysh yes, I will add cases cover this

**Comment by hzxuzhonghu** [comment]
@soltysh done. But I do not know why put fakePodControl in injection.go, which only useful for test. I think it's better to move to cronjob_controller_test.go. What do you think?

**Review by janetkuo — APPROVED** [review]

**Comment by janetkuo** [comment]
This lgtm, @soltysh would you take another look?

**Review by hzxuzhonghu — COMMENTED** [review]
done、

**Comment by hzxuzhonghu** [comment]
kindly ping @soltysh test case covered, would you please review?

File: pkg/controller/cronjob/cronjob_controller.go [hunk 2]
```diff
@@ -376,6 +377,7 @@ func deleteJob(sj *batchv2alpha1.CronJob, job *batchv1.Job, jc jobControlInterfa
 	podList, err := pc.ListPods(job.Namespace, options)
 	if err != nil {
 		recorder.Eventf(sj, v1.EventTypeWarning, "FailedList", "List job-pods: %v", err)
+		return false
```
[the remark this answers is not in the document]
Comment (soltysh) [reply pkg/controller/cronjob/cronjob_controller.go]: Good catch, thx!

**Review by soltysh — APPROVED** [review]

**Pull request merged**